### ISSUE TITLE
Add this to the AssetLib

### ISSUE THREAD

**Issue by darkhog** [Dudejoe870/godot-openmpt#4]
The only mod playing addon there is some shitty made from scratch one that doesn't even support Impulse Tracker. Kinda thought I would have to port libopenmpt myself to godot, fortunately found this.

It sure would be nice if the prebuilt version was available on AssetLib.

**Comment by Dudejoe870** [comment]
I'd love to eventually, but as it doesn't support prebuilt binaries for all platforms, I'm not sure it'd be accepted? It's not something I'm currently working on because I don't have a Mac, and it sounds like a pain to setup the Android SDK and test it and all of that (don't even have an IPhone!). So because of that, I don't think it's in a state that's ready for that. It 100% is here though if anyone wants to volunteer to get it ready for that and do testing (open issues / PRs and I'll take a look!)

**Comment by darkhog** [comment]
I got few plugins from the AssetLib that only have binaries for desktop platforms so you&#39;ll be fine. And since it&#39;s opensource, people will be able to just compile it themselves if they need non-standard binary.   By the way, does this library need any specialized DLL to be distributed along with the game&#39;s executable?  
     
      
       
        Dnia 21 czerwca 2024 21:46 Dudejoe870 ***@***.***&gt; napisał(a):
       
    
       
         
 I&#39;d love to eventually, but as it doesn&#39;t support prebuilt binaries for all platforms, I&#39;m not sure it&#39;d be accepted? It&#39;s not something I&#39;m currently working on because I don&#39;t have a Mac, and it sounds like a pain to setup the Android SDK and test it and all of that (don&#39;t even have an IPhone!). So because of that, I don&#39;t think it&#39;s in a state that&#39;s ready for that. It 100% is here though if anyone wants to volunteer to get it ready for that and do testing (open issues / PRs and I&#39;ll take a look!)

**Comment by GeminiSquishGames** [comment]
The .dll is compiled with the source and is put along side the .exe when you build the godot project so you don't have to do anything else but have the addon in your project.

**Comment by GeminiSquishGames** [comment]
Don't hate me but seeing on how they are pushing for a more FMOD Studio like implementation of interactive music for the 4.3 release I'm heavily suggesting they look at your addon in my proposal and maybe consider the libopenmpt library in general. I really think you should share your work with the rest of the class. For people like me this is god tier stuff and I know it was a big deal in the Unity forums at one point since they made their half ass attempt at tracker support.

**Comment by GeminiSquishGames** [comment]
I'll probably be able to help on the Android front soon, as I have the resources and a bit of time for that. It'll probably be better to put it up though, there isn't a requirement for how many platforms you support. Sorta like how Steam doesn't have a requirement for competent games? lulz. Still having trouble with getting it to build but since 4.3 will by about soon it might help some of the SCons problems and such.

**Comment by DaFlarf** [comment]
This would be an interesting addition to godot. There are already plugins for .mod and .xm in the assetlib, but yours would be the first to support .it. .it would be a really neat format for godot to support because that's what a lot of chiptune conversion tools (SnesMod, VGMmusicmaker) use. Then the doors would be open to making chiptune DAWs or trackers within godot.

TLDR: PLEASE ADD THIS TO THE ASSETLIB I'M BEGGING YOOOOOUUUUUU

**Comment by darkhog** [comment]
@DaFlarf I know what plugins you're referring to and those are really bad at playing module music. Tried using them, but they had problems with some effect codes, which eventually made me set out looking for a better solution, eventually finding this plugin.

**Comment by DaFlarf** [comment]
@darkhog 
Can you walk me through how to add this plugin to a project? I'm not very familiar with weird command line stuff (I've only done so with SNES conversion tools)

**Comment by GeminiSquishGames** [comment]
@DaFlarf 
> @darkhog Can you walk me through how to add this plugin to a project? I'm not very familiar with weird command line stuff (I've only done so with SNES conversion tools)

I got to you first: There should be a release .zip with the already compiled addon in it, just look on the right side of the main page of the git and you should see it under Releases. Click on v1.3, Download and pop it into your addon folder of your Godot Project. Easy peasy no elbow greasy.

**Comment by DaFlarf** [comment]
@GeminiSquishGames That easy huh? I'll give it a shot

**Comment by DaFlarf** [comment]
@GeminiSquishGames 
![Screenshot 2025-01-09 172146](https://github.com/user-attachments/assets/a8955141-9e55-44ea-adce-9b1e00708cdf)
this probably shouldn't be happening

**Comment by GeminiSquishGames** [comment]
@DaFlarf Probably need to go into the addon folder in the .zip and extract that godot-openmpt folder into your addon folder rather than how you have it. It's nested in two other folders called godot-openmpt-prebuilt and that's probably confusing it. I think. 

Just do this:
![image](https://github.com/user-attachments/assets/96951690-466d-470d-8ca8-65c0b1af4fb7)

**Comment by DaFlarf** [comment]
@GeminiSquishGames still getting an error 
![Screenshot 2025-01-10 164435](https://github.com/user-attachments/assets/3b2e441d-51cf-4b7b-ad6a-95aece103b74)

**Comment by GeminiSquishGames** [comment]
@DaFlarf Does it persist if you save and reload the project? I get an error with a new project but it goes away when I "off-on" it.

**Comment by DaFlarf** [comment]
@GeminiSquishGames 
![Screenshot 2025-01-11 123909](https://github.com/user-attachments/assets/5625bc98-3d5d-4b6f-9237-219c9691e71b)
It's working now. Thanks! However, these debug messages are very annoying

**Comment by darkhog** [comment]
I just mute warnings. Only errors matter anyway.

**Comment by DaFlarf** [comment]
@darkhog Lol that's what I did during my game jam.
